src/Client/Suggest.cpp

```diff
@@ -94,7 +95,14 @@ void Suggest::load(ContextPtr context, const ConnectionParameters & connection_p
 {
     loading_thread = std::thread([my_context=Context::createCopy(context), connection_parameters, suggestion_limit, this]
     {
+        /// Creates new QueryScope/ThreadStatus to avoid sharing global context, which settings can be modified by the client in another thread.
         ThreadStatus thread_status;
+        /// LocalConnection creates QueryScope for each query
+        if (!std::is_same_v<ConnectionType, LocalConnection>)
+            CurrentThread::QueryScope query_scope(my_context);
```

Comment:
I assume we doing this just for the side effects. Cause `query_scope` is destroyed after the if statement, and I have not seen a use of `CurrentThread::QueryScope` like this in our codebase. It's always scoped to a much bigger block, a `std::unique_ptr` or even an `std::optional`